Modules/Module_Import/Import/xASL_imp_DetermineSubjectStructure.m

```diff
@@ -153,10 +148,19 @@
         warning('x.SUBJECTS is undefined, data loading is going to fail...');
     end
     
-
+	% If tokenVisitAliases is not provided, we assume visits are not renamed and create a dummy tokenVisitAliases list
+	if ~isfield(x.modules.import.imPar, 'tokenVisitAliases') || isempty(x.modules.import.imPar.tokenVisitAliases)
+		% Create a unique list of sessions
+		listVisits = unique(listVisits);
+
+		% Exclude empty session names
```

Comment:
session -> visit also in this function

Plus, per discussion above, we should not exclude empty visit names right? Instead, we convert them to "missingSessionName"?

Reply:
That's a subfunction for temp-data. The visit-names were renamed to missionSessionValue already. This is a failsafe.

Reply:
1. you didn't react to 
"session -> visit also in this function"
please read again (it means that the comments also need to be visits, or visit/session)

2. I'm confused. Your comment is not sufficiently explainable, so here are again 20 minutes of work :) At the first moment we process tokenVisitAliases in import, notice that it is missing, we recreate it from all existing visit values by saying that they are the same right?

Either this is 
situation A: we work with tempdata here.
In this case, the tokenVisitAlias stuff has been defined already above and this piece of code is unnecessary

situation B: we don't work with tempdata here (I.e. this is before the DICOM-conversion) and we should properly define the dummy tokenVisitAliases.

If you do want a failsafe, I would i) add a warning ii) convert the empty session name to something like 'sessionNameGoneWrong'. Again, in all cases we want everything converted to BIDS. If it is messy, the user can then read this in the BIDS stage. We don't want half of it converted because the other half had i/o issues and the user either a) not noticing that half was converted and reporting incomplete data OR b) has to go through the whole sourcedata folders to know if everything was converted properly or not.

So one important principle: we want to have the user know that everything was converted correctly. So errors or warnings are more important than code speed/failsafes that remove the potential for code crashing because of incorrect data. Of course, this would be in one of the import instances (assuming the import is a module where each subject has a log file).

Reply:
> 1. you didn't react to
>    "session -> visit also in this function"
>    please read again (it means that the comments also need to be visits, or visit/session)

I have changed everything in this function from sessions to visits except for comments that are to the user and talk about BIDS. If you want to change sessions to visits or back, you have to do it yourself cause I have spend 1h on this and I really can't find a way that you would be happy with.

> 2. I'm confused. Your comment is not sufficiently explainable, so here are again 20 minutes of work :) At the first moment we process tokenVisitAliases in import, notice that it is missing, we recreate it from all existing visit values by saying that they are the same right?

Yes

> Either this is situation A: we work with tempdata here. In this case, the tokenVisitAlias stuff has been defined already above and this piece of code is unnecessary

We work with tempdata here. Since the import is modular, we have to make sure that tokenVisitAlias is again correctly initiated as the renaming from  visit to visit alias happens in NII2BIDS and not in DCM2NII.
 
> situation B: we don't work with tempdata here (I.e. this is before the DICOM-conversion) and we should properly define the dummy tokenVisitAliases.
> 
> If you do want a failsafe, I would i) add a warning ii) convert the empty session name to something like 'sessionNameGoneWrong'. Again, in all cases we want everything converted to BIDS. If it is messy, the user can then read this in the BIDS stage. We don't want half of it converted because the other half had i/o issues and the user either a) not noticing that half was converted and reporting incomplete data OR b) has to go through the whole sourcedata folders to know if everything was converted properly or not.
> 
> So one important principle: we want to have the user know that everything was converted correctly. So errors or warnings are more important than code speed/failsafes that remove the potential for code crashing because of incorrect data. Of course, this would be in one of the import instances (assuming the import is a module where each subject has a log file).

So this is just a continuation of the previous stuff and the user warning was already issued.

Reply:
> 1. Line 153:
>    `	% Create a unique list of sessions`
>    `		listVisits = unique(listVisits);`
>    `		% Exclude empty session names`
> 
> so "all was not changed already" (or am I seeing something wrong?)

Yes - you are looking at the original comment - which of course displays the original commit (see the "outdated" label). This has been modified in between, but this is only displayed in the list file changes, and won't be altered here.

> -> the temp step is an arbitrary in-between step, should not be "modular" necessarily. We should not need to re-initiate tokenVisitAlias. Either we manage the tokenVisitAlias fully at the start, or at the temp stage. Not partly at the start and partly at the temp stage. But I guess this is something we need to do properly in the next issue.

You probably don't remember this. But this is your decision. We wanted to have a joint/semi-separate DCM2NII and NII2BIDS. But you strongly insisted that these two things have to be completely separated and fully modular. That took as a lot of work to separate them completely, but it was you who strongly insisted that we should do all the initializations again and from scratch. So you can't now complain about this ;)

> It still shows me that it excludes empty session names, whereas you say that these are replaced by missingSessionValues (or here it should be 'sessionNameGoneWrong' -> if you read that part ;)
> 
> Do we miscommunicate because I see old code?

Yes - here you of course only see the old code. See the label "outdated", it won't reinitialize this code excerpt.

Reply:
I see your point. Then tokenVisitAlias should neither be initiated/managed in dcm2nii OR in NII2BIDS, but in the general import initialization right?

-> How do I see the new code with the old comment then? If I want to check if code is changed according to a comment (that is the whole point of seeing comments "inline")